### PR TITLE
Converter Winding

### PR DESCRIPTION
**Pull request by clementd08**
Hello all,

This pull request introduce a new part for converter, conversion for winding.

Best regards,
Clément

### CODE REVIEW

**Comment by clementd08** [comment]
Check conversion for end_winding_lenght

File: pyleecan/Functions/Converter/MotorCAD/end_winding_lenght.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    MLT = self.get_other(other_dict, other_path_list, other_unit_dict)
+
+    if machine.stator.winding.end_winding is None:
+        cp = 0
+    else:
+        cp = machine.stator.winding.end_winding.comp_length()
+
+    lewout = MLT / 2 - machine.stator.comp_length() - cp
+
+    machine.stator.winding.Lewout = lewout / 1000  # convertion mm to m
```
Comment (BonneelP) [inline pyleecan/Functions/Converter/MotorCAD/end_winding_lenght.py:31]: Check unit

File: pyleecan/Functions/Converter/MotorCAD/end_winding_lenght.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    self.unit_type = ""
+    other_path_list = ["[Winding_Design]", "EWdg_MLT"]
+    MLT = self.get_other(other_dict, other_path_list, other_unit_dict)
+
+    if machine.stator.winding.end_winding is None:
+        cp = 0
+    else:
+        cp = machine.stator.winding.end_winding.comp_length()
+
+    lewout = MLT / 2 - machine.stator.comp_length() - cp
```
Comment (BonneelP) [inline pyleecan/Functions/Converter/MotorCAD/end_winding_lenght.py:29]: Check formula

File: pyleecan/Methods/Converter/Convert/Step/select_conductor_rules.py [hunk 3]
```diff
@@ -9,4 +14,21 @@ def select_conductor_rules(self, is_stator):
...
+        conductor = self.machine.stator.winding.conductor
+    else:
+        conductor = self.machine.rotor.winding.conductor
+
+    # add the correct rule depending on the rotor
+    if isinstance(conductor, CondType12):
+        self.add_rule_condtype12(is_stator)
+
+    elif isinstance(conductor, CondType11):
+        self.add_rule_condtype11(is_stator)
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/Convert/Step/select_conductor_rules.py:34]: else raise error unknown error

File: pyleecan/Methods/Converter/Convert/Step/select_conductor_rules.py [hunk 4]
```diff
@@ -1,3 +1,8 @@
+from .....Classes.CondType12 import CondType12
+from .....Classes.CondType11 import CondType11
+from .....Classes.MachineWRSM import MachineWRSM
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/Convert/Step/select_conductor_rules.py:3]: remove WRSM import

File: pyleecan/Methods/Converter/Convert/Step/select_winding_rules.py [hunk 5]
```diff
@@ -7,6 +10,14 @@ def select_winding_rules(self, is_stator):
         A ConvertMC object
     is_stator : bool
         True slot is in stator, False slot is in rotor
-
     """
-    self.get_logger().info("winding, not implemented")
+
+    if isinstance(self.machine, MachineWRSM):
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/Convert/Step/select_winding_rules.py:15]: What about WRSM stator

File: pyleecan/Methods/Converter/ConvertMC/Rules/Winding/add_rule_condtype12.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                    "variable": "b",
+                },
+                {
+                    "src": "pyleecan",
+                    "path": f"machine.{lam_name}.winding.conductor.Wins_wire",
+                    "variable": "x",
+                },
+            ],
+            unit_type="m",
+            equation="y/2 + b = x",
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/ConvertMC/Rules/Winding/add_rule_condtype12.py:63]: y/2 + b/2 = x

File: pyleecan/Methods/Converter/ConvertMC/Rules/Winding/add_rule_condtype11.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    is_stator : Bool
+        True slot is in stator, False slot is in rotor
+    """
+    # definie the correct position in rotor or in stator
+    if is_stator == True:
+        lam_name = "stator"
+    else:
+        lam_name = "rotor"
+
+    """
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/ConvertMC/Rules/Winding/add_rule_condtype11.py:23]: Add comment on why

File: pyleecan/Methods/Converter/ConvertMC/convert_to_P/convert_conductor_to_P.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        A ConvertMC object
+    """
+    if is_stator:
+        winding = self.machine.stator.winding
+    else:
+        winding = self.machine.rotor.winding
+
+    conductor_type = self.other_dict["[Winding_Design]"]["Wire_Type"]
+
+    if conductor_type in ["AWG_Table", "Metric_Table", "Diameter_Input", "SWG_Table"]:
```
Comment (BonneelP) [inline pyleecan/Methods/Converter/ConvertMC/convert_to_P/convert_conductor_to_P.py:20]: motor cad schematics to check

**Pull request merged**